app.js

```diff
@@ -0,0 +1,207 @@
+// this is the game board
+const gameBoard = () => {
+
+    const _board = [['','',''],['','',''],['','','']];
+
+    const setMark = (mark,i,j) => {
+      if(_board[i][j].length > 0){
+        return false;
+      }else{
+        _board[i][j] = mark;
+        return true;
+      }
+    }
+
+const checkIfFull = () => {
+          let isFull = true;
+          _board.forEach(row => {
+            row.forEach(cell => {
+                if(cell.length == 0){
+                  isFull = false;
+                }
+            })
+          })
+          return isFull;
+        }
+
+    const getBoard = () => _board;
+
+    const checkForWinner = (mark)=> {
+
+      // horizontal
+      if(_board[0][0] == mark && _board[0][1] == mark && _board[0][2] == mark){ return true; }
+      if(_board[1][0] == mark && _board[1][1] == mark && _board[1][2] == mark){ return true; }
+      if(_board[2][0] == mark && _board[2][1] == mark && _board[2][2] == mark){ return true; }
+
+      // vertical
+      if(_board[0][0] == mark && _board[1][0] == mark && _board[2][0] == mark){ return true; }
+      if(_board[0][1] == mark && _board[1][1] == mark && _board[2][1] == mark){ return true; }
+      if(_board[0][2] == mark && _board[1][2] == mark && _board[2][2] == mark){ return true; }
+
+      // diagonal
+      if(_board[0][0] == mark && _board[1][1] == mark && _board[2][2] == mark){ return true; }
+      if(_board[0][2] == mark && _board[1][1] == mark && _board[2][0] == mark){ return true; }
+
+      return false;
+    }
+
+    return { getBoard, setMark, checkIfFull,checkForWinner };
+}
+const playerFactory = (name, mark) => {
+    const getName = () => name;
+    const getMark = () => mark;
+    const setName = (n) => { name = n;}
+    return{ getName, getMark, setName };
```

Comment:
I highly recommend to give `player` something to do. Please refer to my annotation/suggestion in the `board` factory on how you can set up your player to work with the architecture I suggested there.

Reply:
We have added a method in player factory play which calls on board method setmark and increments player moves so we can check it against wining combos.